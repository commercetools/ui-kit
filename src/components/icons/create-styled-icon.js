import PropTypes from 'prop-types';
import invariant from 'tiny-invariant';
import styled from '@emotion/styled';
import vars from '../../../materials/custom-properties';

const iconSizes = {
  small: 12,
  medium: 16,
  big: 24,
};

const getSizeStyle = size => {
  switch (size) {
    case 'scale':
      return `
        &:not(:root) {
          width: 100%;
          height: auto;
        }
      `;
    case 'small':
    case 'medium':
    case 'big':
      return `
        width: ${iconSizes[size]}px;
        height: ${iconSizes[size]}px;
      `;
    default:
      return `
        width: ${iconSizes.big}px;
        height: ${iconSizes.big}px;
      `;
  }
};

const getColor = theme => {
  if (!theme) return 'inherit';
  switch (theme) {
    case 'black':
      return vars.colorBlack;
    case 'grey':
      return vars.colorGray60;
    case 'white':
      return vars.colorWhite;
    case 'blue':
      return vars.colorBlue;
    case 'green':
      return vars.colorGreen;
    case 'green-light':
      return vars.colorGreen40;
    case 'orange':
      return vars.colorOrange;
    case 'red':
      return vars.colorRed;
    default: {
      invariant(
        theme,
        `ui-kit/Icon: the specified theme '${theme}' is not supported.`
      );
      return 'inherit';
    }
  }
};

export default function createStyledIcon(Component, displayName) {
  const StyledComponent = styled(Component)(
    props => `
    * {
      fill: ${getColor(props.theme)};
    }
    ${getSizeStyle(props.size)}
  `
  );
  StyledComponent.displayName = displayName;
  StyledComponent.propTypes = {
    theme: PropTypes.oneOf([
      'black',
      'grey',
      'white',
      'blue',
      'green',
      'green-light',
      'orange',
      'red',
    ]),
    size: PropTypes.oneOf(['small', 'medium', 'big', 'scale']),
  };
  return StyledComponent;
}
