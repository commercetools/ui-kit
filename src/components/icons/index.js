// This file is auto-generated using the 'generate-icon-exports.js' script
// so any changes made to this file manually will be lost the next time the
// script is executed
import createStyledIcon from './create-styled-icon';
import OrigAngleDownIcon from './svg/angle-down.react.svg';
import OrigAngleLeftIcon from './svg/angle-left.react.svg';
import OrigAngleRightIcon from './svg/angle-right.react.svg';
import OrigAngleThinLeftIcon from './svg/angle-thin-left.react.svg';
import OrigAngleThinRightIcon from './svg/angle-thin-right.react.svg';
import OrigAngleUpIcon from './svg/angle-up.react.svg';
import OrigArrowDownIcon from './svg/arrow-down.react.svg';
import OrigArrowLeftIcon from './svg/arrow-left.react.svg';
import OrigArrowLongDownIcon from './svg/arrow-long-down.react.svg';
import OrigArrowRightIcon from './svg/arrow-right.react.svg';
import OrigArrowTriangleDownIcon from './svg/arrow-triangle-down.react.svg';
import OrigArrowTriangleUpIcon from './svg/arrow-triangle-up.react.svg';
import OrigArrowUpIcon from './svg/arrow-up.react.svg';
import OrigArrowsIcon from './svg/arrows.react.svg';
import OrigBackIcon from './svg/back.react.svg';
import OrigBidirectionalArrowIcon from './svg/bidirectional-arrow.react.svg';
import OrigBinFilledIcon from './svg/bin-filled.react.svg';
import OrigBinLinearIcon from './svg/bin-linear.react.svg';
import OrigBoxIcon from './svg/box.react.svg';
import OrigBrainIcon from './svg/brain.react.svg';
import OrigCalendarIcon from './svg/calendar.react.svg';
import OrigCameraIcon from './svg/camera.react.svg';
import OrigCaretDownIcon from './svg/caret-down.react.svg';
import OrigCaretUpIcon from './svg/caret-up.react.svg';
import OrigCartIcon from './svg/cart.react.svg';
import OrigChainBrokenIcon from './svg/chain-broken.react.svg';
import OrigChainIcon from './svg/chain.react.svg';
import OrigCheckActiveIcon from './svg/check-active.react.svg';
import OrigCheckBoldIcon from './svg/check-bold.react.svg';
import OrigCheckInactiveIcon from './svg/check-inactive.react.svg';
import OrigCheckThinIcon from './svg/check-thin.react.svg';
import OrigCircleIcon from './svg/circle.react.svg';
import OrigClipboardIcon from './svg/clipboard.react.svg';
import OrigClockIcon from './svg/clock.react.svg';
import OrigCloseBoldIcon from './svg/close-bold.react.svg';
import OrigCloseIcon from './svg/close.react.svg';
import OrigCodeViewIcon from './svg/code-view.react.svg';
import OrigCoinsIcon from './svg/coins.react.svg';
import OrigColumnsIcon from './svg/columns.react.svg';
import OrigConnectedSquareIcon from './svg/connected-square.react.svg';
import OrigConnectedTriangleIcon from './svg/connected-triangle.react.svg';
import OrigCopyIcon from './svg/copy.react.svg';
import OrigCubeIcon from './svg/cube.react.svg';
import OrigCubesIcon from './svg/cubes.react.svg';
import OrigDotIcon from './svg/dot.react.svg';
import OrigDownloadIcon from './svg/download.react.svg';
import OrigDragDropIcon from './svg/drag-drop.react.svg';
import OrigDragIcon from './svg/drag.react.svg';
import OrigEditIcon from './svg/edit.react.svg';
import OrigErrorIcon from './svg/error.react.svg';
import OrigExpandIcon from './svg/expand.react.svg';
import OrigExportIcon from './svg/export.react.svg';
import OrigExternalLinkIcon from './svg/external-link.react.svg';
import OrigEyeCrossedIcon from './svg/eye-crossed.react.svg';
import OrigEyeIcon from './svg/eye.react.svg';
import OrigFilterIcon from './svg/filter.react.svg';
import OrigFlagFilledIcon from './svg/flag-filled.react.svg';
import OrigFlagLinearIcon from './svg/flag-linear.react.svg';
import OrigFlameIcon from './svg/flame.react.svg';
import OrigGearIcon from './svg/gear.react.svg';
import OrigGraduationCapIcon from './svg/graduation-cap.react.svg';
import OrigGraphIcon from './svg/graph.react.svg';
import OrigGridIcon from './svg/grid.react.svg';
import OrigHeartIcon from './svg/heart.react.svg';
import OrigImportIcon from './svg/import.react.svg';
import OrigInfoIcon from './svg/info.react.svg';
import OrigInformationIcon from './svg/information.react.svg';
import OrigListIcon from './svg/list.react.svg';
import OrigLogoutIcon from './svg/logout.react.svg';
import OrigMailIcon from './svg/mail.react.svg';
import OrigMinimizeIcon from './svg/minimize.react.svg';
import OrigNestedViewIcon from './svg/nested-view.react.svg';
import OrigPageGearIcon from './svg/page-gear.react.svg';
import OrigPagesIcon from './svg/pages.react.svg';
import OrigPaperBillInvertedIcon from './svg/paper-bill-inverted.react.svg';
import OrigPaperclipIcon from './svg/paperclip.react.svg';
import OrigPinFilledIcon from './svg/pin-filled.react.svg';
import OrigPinLinearIcon from './svg/pin-linear.react.svg';
import OrigPluginIcon from './svg/plugin.react.svg';
import OrigPlusBoldIcon from './svg/plus-bold.react.svg';
import OrigPlusThinIcon from './svg/plus-thin.react.svg';
import OrigRefreshIcon from './svg/refresh.react.svg';
import OrigRestoreIcon from './svg/restore.react.svg';
import OrigRevertIcon from './svg/revert.react.svg';
import OrigReviewIcon from './svg/review.react.svg';
import OrigRocketIcon from './svg/rocket.react.svg';
import OrigScreenGearIcon from './svg/screen-gear.react.svg';
import OrigScreenUserIcon from './svg/screen-user.react.svg';
import OrigSearchIcon from './svg/search.react.svg';
import OrigSortingIcon from './svg/sorting.react.svg';
import OrigSpeechBubbleIcon from './svg/speech-bubble.react.svg';
import OrigSpeedometerIcon from './svg/speedometer.react.svg';
import OrigSplitIcon from './svg/split.react.svg';
import OrigStackIcon from './svg/stack.react.svg';
import OrigStarIcon from './svg/star.react.svg';
import OrigSupportIcon from './svg/support.react.svg';
import OrigSwitcherIcon from './svg/switcher.react.svg';
import OrigTableIcon from './svg/table.react.svg';
import OrigTagMultiIcon from './svg/tag-multi.react.svg';
import OrigTagStackedIcon from './svg/tag-stacked.react.svg';
import OrigTagIcon from './svg/tag.react.svg';
import OrigTerminalIcon from './svg/terminal.react.svg';
import OrigTreeStructureIcon from './svg/tree-structure.react.svg';
import OrigTruckIcon from './svg/truck.react.svg';
import OrigUserFilledIcon from './svg/user-filled.react.svg';
import OrigUserLinearIcon from './svg/user-linear.react.svg';
import OrigVerifiedIcon from './svg/verified.react.svg';
import OrigWarningIcon from './svg/warning.react.svg';
import OrigWorldIcon from './svg/world.react.svg';

export const AngleDownIcon = createStyledIcon(
  OrigAngleDownIcon,
  'AngleDownIcon'
);
export const AngleLeftIcon = createStyledIcon(
  OrigAngleLeftIcon,
  'AngleLeftIcon'
);
export const AngleRightIcon = createStyledIcon(
  OrigAngleRightIcon,
  'AngleRightIcon'
);
export const AngleThinLeftIcon = createStyledIcon(
  OrigAngleThinLeftIcon,
  'AngleThinLeftIcon'
);
export const AngleThinRightIcon = createStyledIcon(
  OrigAngleThinRightIcon,
  'AngleThinRightIcon'
);
export const AngleUpIcon = createStyledIcon(OrigAngleUpIcon, 'AngleUpIcon');
export const ArrowDownIcon = createStyledIcon(
  OrigArrowDownIcon,
  'ArrowDownIcon'
);
export const ArrowLeftIcon = createStyledIcon(
  OrigArrowLeftIcon,
  'ArrowLeftIcon'
);
export const ArrowLongDownIcon = createStyledIcon(
  OrigArrowLongDownIcon,
  'ArrowLongDownIcon'
);
export const ArrowRightIcon = createStyledIcon(
  OrigArrowRightIcon,
  'ArrowRightIcon'
);
export const ArrowTriangleDownIcon = createStyledIcon(
  OrigArrowTriangleDownIcon,
  'ArrowTriangleDownIcon'
);
export const ArrowTriangleUpIcon = createStyledIcon(
  OrigArrowTriangleUpIcon,
  'ArrowTriangleUpIcon'
);
export const ArrowUpIcon = createStyledIcon(OrigArrowUpIcon, 'ArrowUpIcon');
export const ArrowsIcon = createStyledIcon(OrigArrowsIcon, 'ArrowsIcon');
export const BackIcon = createStyledIcon(OrigBackIcon, 'BackIcon');
export const BidirectionalArrowIcon = createStyledIcon(
  OrigBidirectionalArrowIcon,
  'BidirectionalArrowIcon'
);
export const BinFilledIcon = createStyledIcon(
  OrigBinFilledIcon,
  'BinFilledIcon'
);
export const BinLinearIcon = createStyledIcon(
  OrigBinLinearIcon,
  'BinLinearIcon'
);
export const BoxIcon = createStyledIcon(OrigBoxIcon, 'BoxIcon');
export const BrainIcon = createStyledIcon(OrigBrainIcon, 'BrainIcon');
export const CalendarIcon = createStyledIcon(OrigCalendarIcon, 'CalendarIcon');
export const CameraIcon = createStyledIcon(OrigCameraIcon, 'CameraIcon');
export const CaretDownIcon = createStyledIcon(
  OrigCaretDownIcon,
  'CaretDownIcon'
);
export const CaretUpIcon = createStyledIcon(OrigCaretUpIcon, 'CaretUpIcon');
export const CartIcon = createStyledIcon(OrigCartIcon, 'CartIcon');
export const ChainBrokenIcon = createStyledIcon(
  OrigChainBrokenIcon,
  'ChainBrokenIcon'
);
export const ChainIcon = createStyledIcon(OrigChainIcon, 'ChainIcon');
export const CheckActiveIcon = createStyledIcon(
  OrigCheckActiveIcon,
  'CheckActiveIcon'
);
export const CheckBoldIcon = createStyledIcon(
  OrigCheckBoldIcon,
  'CheckBoldIcon'
);
export const CheckInactiveIcon = createStyledIcon(
  OrigCheckInactiveIcon,
  'CheckInactiveIcon'
);
export const CheckThinIcon = createStyledIcon(
  OrigCheckThinIcon,
  'CheckThinIcon'
);
export const CircleIcon = createStyledIcon(OrigCircleIcon, 'CircleIcon');
export const ClipboardIcon = createStyledIcon(
  OrigClipboardIcon,
  'ClipboardIcon'
);
export const ClockIcon = createStyledIcon(OrigClockIcon, 'ClockIcon');
export const CloseBoldIcon = createStyledIcon(
  OrigCloseBoldIcon,
  'CloseBoldIcon'
);
export const CloseIcon = createStyledIcon(OrigCloseIcon, 'CloseIcon');
export const CodeViewIcon = createStyledIcon(OrigCodeViewIcon, 'CodeViewIcon');
export const CoinsIcon = createStyledIcon(OrigCoinsIcon, 'CoinsIcon');
export const ColumnsIcon = createStyledIcon(OrigColumnsIcon, 'ColumnsIcon');
export const ConnectedSquareIcon = createStyledIcon(
  OrigConnectedSquareIcon,
  'ConnectedSquareIcon'
);
export const ConnectedTriangleIcon = createStyledIcon(
  OrigConnectedTriangleIcon,
  'ConnectedTriangleIcon'
);
export const CopyIcon = createStyledIcon(OrigCopyIcon, 'CopyIcon');
export const CubeIcon = createStyledIcon(OrigCubeIcon, 'CubeIcon');
export const CubesIcon = createStyledIcon(OrigCubesIcon, 'CubesIcon');
export const DotIcon = createStyledIcon(OrigDotIcon, 'DotIcon');
export const DownloadIcon = createStyledIcon(OrigDownloadIcon, 'DownloadIcon');
export const DragDropIcon = createStyledIcon(OrigDragDropIcon, 'DragDropIcon');
export const DragIcon = createStyledIcon(OrigDragIcon, 'DragIcon');
export const EditIcon = createStyledIcon(OrigEditIcon, 'EditIcon');
export const ErrorIcon = createStyledIcon(OrigErrorIcon, 'ErrorIcon');
export const ExpandIcon = createStyledIcon(OrigExpandIcon, 'ExpandIcon');
export const ExportIcon = createStyledIcon(OrigExportIcon, 'ExportIcon');
export const ExternalLinkIcon = createStyledIcon(
  OrigExternalLinkIcon,
  'ExternalLinkIcon'
);
export const EyeCrossedIcon = createStyledIcon(
  OrigEyeCrossedIcon,
  'EyeCrossedIcon'
);
export const EyeIcon = createStyledIcon(OrigEyeIcon, 'EyeIcon');
export const FilterIcon = createStyledIcon(OrigFilterIcon, 'FilterIcon');
export const FlagFilledIcon = createStyledIcon(
  OrigFlagFilledIcon,
  'FlagFilledIcon'
);
export const FlagLinearIcon = createStyledIcon(
  OrigFlagLinearIcon,
  'FlagLinearIcon'
);
export const FlameIcon = createStyledIcon(OrigFlameIcon, 'FlameIcon');
export const GearIcon = createStyledIcon(OrigGearIcon, 'GearIcon');
export const GraduationCapIcon = createStyledIcon(
  OrigGraduationCapIcon,
  'GraduationCapIcon'
);
export const GraphIcon = createStyledIcon(OrigGraphIcon, 'GraphIcon');
export const GridIcon = createStyledIcon(OrigGridIcon, 'GridIcon');
export const HeartIcon = createStyledIcon(OrigHeartIcon, 'HeartIcon');
export const ImportIcon = createStyledIcon(OrigImportIcon, 'ImportIcon');
export const InfoIcon = createStyledIcon(OrigInfoIcon, 'InfoIcon');
export const InformationIcon = createStyledIcon(
  OrigInformationIcon,
  'InformationIcon'
);
export const ListIcon = createStyledIcon(OrigListIcon, 'ListIcon');
export const LogoutIcon = createStyledIcon(OrigLogoutIcon, 'LogoutIcon');
export const MailIcon = createStyledIcon(OrigMailIcon, 'MailIcon');
export const MinimizeIcon = createStyledIcon(OrigMinimizeIcon, 'MinimizeIcon');
export const NestedViewIcon = createStyledIcon(
  OrigNestedViewIcon,
  'NestedViewIcon'
);
export const PageGearIcon = createStyledIcon(OrigPageGearIcon, 'PageGearIcon');
export const PagesIcon = createStyledIcon(OrigPagesIcon, 'PagesIcon');
export const PaperBillInvertedIcon = createStyledIcon(
  OrigPaperBillInvertedIcon,
  'PaperBillInvertedIcon'
);
export const PaperclipIcon = createStyledIcon(
  OrigPaperclipIcon,
  'PaperclipIcon'
);
export const PinFilledIcon = createStyledIcon(
  OrigPinFilledIcon,
  'PinFilledIcon'
);
export const PinLinearIcon = createStyledIcon(
  OrigPinLinearIcon,
  'PinLinearIcon'
);
export const PluginIcon = createStyledIcon(OrigPluginIcon, 'PluginIcon');
export const PlusBoldIcon = createStyledIcon(OrigPlusBoldIcon, 'PlusBoldIcon');
export const PlusThinIcon = createStyledIcon(OrigPlusThinIcon, 'PlusThinIcon');
export const RefreshIcon = createStyledIcon(OrigRefreshIcon, 'RefreshIcon');
export const RestoreIcon = createStyledIcon(OrigRestoreIcon, 'RestoreIcon');
export const RevertIcon = createStyledIcon(OrigRevertIcon, 'RevertIcon');
export const ReviewIcon = createStyledIcon(OrigReviewIcon, 'ReviewIcon');
export const RocketIcon = createStyledIcon(OrigRocketIcon, 'RocketIcon');
export const ScreenGearIcon = createStyledIcon(
  OrigScreenGearIcon,
  'ScreenGearIcon'
);
export const ScreenUserIcon = createStyledIcon(
  OrigScreenUserIcon,
  'ScreenUserIcon'
);
export const SearchIcon = createStyledIcon(OrigSearchIcon, 'SearchIcon');
export const SortingIcon = createStyledIcon(OrigSortingIcon, 'SortingIcon');
export const SpeechBubbleIcon = createStyledIcon(
  OrigSpeechBubbleIcon,
  'SpeechBubbleIcon'
);
export const SpeedometerIcon = createStyledIcon(
  OrigSpeedometerIcon,
  'SpeedometerIcon'
);
export const SplitIcon = createStyledIcon(OrigSplitIcon, 'SplitIcon');
export const StackIcon = createStyledIcon(OrigStackIcon, 'StackIcon');
export const StarIcon = createStyledIcon(OrigStarIcon, 'StarIcon');
export const SupportIcon = createStyledIcon(OrigSupportIcon, 'SupportIcon');
export const SwitcherIcon = createStyledIcon(OrigSwitcherIcon, 'SwitcherIcon');
export const TableIcon = createStyledIcon(OrigTableIcon, 'TableIcon');
export const TagMultiIcon = createStyledIcon(OrigTagMultiIcon, 'TagMultiIcon');
export const TagStackedIcon = createStyledIcon(
  OrigTagStackedIcon,
  'TagStackedIcon'
);
export const TagIcon = createStyledIcon(OrigTagIcon, 'TagIcon');
export const TerminalIcon = createStyledIcon(OrigTerminalIcon, 'TerminalIcon');
export const TreeStructureIcon = createStyledIcon(
  OrigTreeStructureIcon,
  'TreeStructureIcon'
);
export const TruckIcon = createStyledIcon(OrigTruckIcon, 'TruckIcon');
export const UserFilledIcon = createStyledIcon(
  OrigUserFilledIcon,
  'UserFilledIcon'
);
export const UserLinearIcon = createStyledIcon(
  OrigUserLinearIcon,
  'UserLinearIcon'
);
export const VerifiedIcon = createStyledIcon(OrigVerifiedIcon, 'VerifiedIcon');
export const WarningIcon = createStyledIcon(OrigWarningIcon, 'WarningIcon');
export const WorldIcon = createStyledIcon(OrigWorldIcon, 'WorldIcon');
