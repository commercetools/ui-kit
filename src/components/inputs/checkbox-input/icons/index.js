// NOTE: explicit names for component names
import CheckboxDefaultIcon from './checkbox-default.react.svg';
import CheckboxCheckedIcon from './checkbox-checked.react.svg';
import CheckboxIndeterminateIcon from './checkbox-indeterminate.react.svg';

export default {
  Unchecked: CheckboxDefaultIcon,
  Checked: CheckboxCheckedIcon,
  Indeterminate: CheckboxIndeterminateIcon,
};
