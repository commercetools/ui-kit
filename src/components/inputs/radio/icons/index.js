// NOTE: explicit names for component names
import RadioOptionDefaultIcon from './radio-option-default.react.svg';
import RadioOptionCheckedIcon from './radio-option-checked.react.svg';

export default {
  Default: RadioOptionDefaultIcon,
  Checked: RadioOptionCheckedIcon,
};
