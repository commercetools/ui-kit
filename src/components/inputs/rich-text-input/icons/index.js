import createStyledIcon from '../../../icons/create-styled-icon';
import OrigBoldIcon from './svg/bold.react.svg';
import OrigItalicIcon from './svg/italic.react.svg';
import OrigOrderedListIcon from './svg/ordered-list.react.svg';
import OrigMoreStylesIcon from './svg/more-styles.react.svg';
import OrigUnderlineIcon from './svg/underline.react.svg';
import OrigUnorderedListIcon from './svg/unordered-list.react.svg';
import OrigStrikethroughIcon from './svg/strikethrough.react.svg';
import OrigSuperscriptIcon from './svg/superscript.react.svg';
import OrigSubscriptIcon from './svg/subscript.react.svg';
import OrigRedoIcon from './svg/redo.react.svg';

export const BoldIcon = createStyledIcon(OrigBoldIcon, 'BoldIcon');
export const ItalicIcon = createStyledIcon(OrigItalicIcon, 'ItalicIcon');
export const MoreStylesIcon = createStyledIcon(
  OrigMoreStylesIcon,
  'MoreStylesIcon'
);
export const OrderedListIcon = createStyledIcon(
  OrigOrderedListIcon,
  'OrderedListIcon'
);
export const UnderlineIcon = createStyledIcon(
  OrigUnderlineIcon,
  'UnderlineIcon'
);
export const UnorderedListIcon = createStyledIcon(
  OrigUnorderedListIcon,
  'UnorderedListIcon'
);

export const StrikethroughIcon = createStyledIcon(
  OrigStrikethroughIcon,
  'StrikethroughIcon'
);
export const SuperscriptIcon = createStyledIcon(
  OrigSuperscriptIcon,
  'SuperscriptIcon'
);
export const SubscriptIcon = createStyledIcon(
  OrigSubscriptIcon,
  'SubscriptIcon'
);

export const RedoIcon = createStyledIcon(OrigRedoIcon, 'RedoIcon');
