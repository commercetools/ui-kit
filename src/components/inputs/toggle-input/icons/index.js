import ToggleButtonIcon from './toggle-button-default.react.svg';
import ToggleBarIcon from './toggle-bar-default.react.svg';

export default {
  ButtonDefault: ToggleButtonIcon,
  BarDefault: ToggleBarIcon,
};
