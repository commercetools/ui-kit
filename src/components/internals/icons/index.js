// This file is auto-generated using the 'generate-icon-exports.js' script
// so any changes made to this file manually will be lost the next time the
// script is executed
export { default as BoldIcon } from './components/bold';
export { default as CheckedIcon } from './components/checked';
export { default as ExpandIcon } from './components/expand';
export { default as IndeterminateIcon } from './components/indeterminate';
export { default as ItalicIcon } from './components/italic';
export { default as MoreStylesIcon } from './components/more-styles';
export { default as OrderedListIcon } from './components/ordered-list';
export { default as RadioOptionCheckedIcon } from './components/radio-option-checked';
export { default as RadioOptionUncheckedIcon } from './components/radio-option-unchecked';
export { default as RedoIcon } from './components/redo';
export { default as StrikethroughIcon } from './components/strikethrough';
export { default as SubscriptIcon } from './components/subscript';
export { default as SuperscriptIcon } from './components/superscript';
export { default as UncheckedIcon } from './components/unchecked';
export { default as UnderlineIcon } from './components/underline';
export { default as UndoIcon } from './components/undo';
export { default as UnorderedListIcon } from './components/unordered-list';
