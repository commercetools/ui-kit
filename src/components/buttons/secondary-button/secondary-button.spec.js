import React from 'react';
import { render, fireEvent, wait } from '../../../test-utils';
import { PlusBoldIcon } from '../../icons';
import SecondaryButton from './secondary-button';

const createTestProps = custom => ({
  label: 'Add',
  iconLeft: <PlusBoldIcon data-testid="icon" />,
  onClick: jest.fn(),
  ...custom,
});

describe('rendering', () => {
  let props;
  beforeEach(() => {
    props = createTestProps();
  });
  it('should render', () => {
    const { getByLabelText } = render(<SecondaryButton {...props} />);
    expect(getByLabelText('Add')).toBeInTheDocument();
    expect(getByLabelText('Add')).not.toHaveAttribute('disabled');
  });
  it('should render icon', () => {
    const { getByTestId } = render(<SecondaryButton {...props} />);
    expect(getByTestId('icon')).toBeInTheDocument();
  });
  it('should not render icon', () => {
    const { queryByTestId } = render(
      <SecondaryButton {...props} iconLeft={undefined} />
    );
    expect(queryByTestId('icon')).not.toBeInTheDocument();
  });
  it('should be marked as "disabled"', () => {
    const { getByLabelText } = render(
      <SecondaryButton {...props} isDisabled={true} />
    );
    expect(getByLabelText('Add')).toHaveAttribute('disabled');
    expect(getByLabelText('Add')).toHaveAttribute('aria-disabled', 'true');
  });
  it('should be marked as "active"', () => {
    const { getByLabelText } = render(
      <SecondaryButton {...props} isToggleButton={true} isToggled={true} />
    );
    expect(getByLabelText('Add')).toHaveAttribute('aria-pressed', 'true');
  });
  describe('type variations', () => {
    it('should render a button of type "button"', () => {
      const { getByLabelText } = render(<SecondaryButton {...props} />);
      expect(getByLabelText('Add')).toHaveAttribute('type', 'button');
    });
    it('should render a button of type "submit"', () => {
      const { getByLabelText } = render(
        <SecondaryButton {...props} type="submit" />
      );
      expect(getByLabelText('Add')).toHaveAttribute('type', 'submit');
    });
    it('should render a button of type "reset"', () => {
      const { getByLabelText } = render(
        <SecondaryButton {...props} type="reset" />
      );
      expect(getByLabelText('Add')).toHaveAttribute('type', 'reset');
    });
  });
  describe('when using "linkTo"', () => {
    it('should navigate to link when clicked', async () => {
      const { getByLabelText, history } = render(
        <SecondaryButton {...props} linkTo="/foo/bar" />
      );
      fireEvent.click(getByLabelText('Add'));
      await wait(() => {
        expect(history.location.pathname).toBe('/foo/bar');
      });
    });
  });
});
