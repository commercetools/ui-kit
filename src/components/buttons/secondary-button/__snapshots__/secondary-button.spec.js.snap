// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering structure should output correct tree 1`] = `
<Div
  className="container button-appearance default theme-default"
  onMouseDown={[MockFunction]}
  onMouseOut={[MockFunction]}
  onMouseOver={[MockFunction]}
  onMouseUp={[MockFunction]}
>
  <AccessibleButton
    buttonAttributes={
      Object {
        "data-track-component": "SecondaryButton",
      }
    }
    className="button"
    isDisabled={false}
    isToggleButton={false}
    label=""
    onClick={[Function]}
    type="button"
  >
    <Inline
      alignItems="center"
      scale="xs"
    >
      <span />
    </Inline>
  </AccessibleButton>
</Div>
`;

exports[`rendering with icon should output correct tree 1`] = `
<Div
  className="container button-appearance default theme-default"
  onMouseDown={[MockFunction]}
  onMouseOut={[MockFunction]}
  onMouseOver={[MockFunction]}
  onMouseUp={[MockFunction]}
>
  <AccessibleButton
    buttonAttributes={
      Object {
        "data-track-component": "SecondaryButton",
      }
    }
    className="button"
    isDisabled={false}
    isToggleButton={false}
    label=""
    onClick={[Function]}
    type="button"
  >
    <Inline
      alignItems="center"
      scale="xs"
    >
      <span
        className="icon-container"
      >
        <FilterIcon />
      </span>
      <span />
    </Inline>
  </AccessibleButton>
</Div>
`;
